src/coreclr/jit/morph.cpp

```diff
@@ -5465,11 +5465,21 @@ GenTree* Compiler::fgMorphArrayIndex(GenTree* tree)
 
     GenTree* cns = gtNewIconNode(elemOffs, TYP_I_IMPL);
 
+// We're going to create a platform-specific address:
+//
+// XArch:   "arrRef + (index + elemOffset)"
+// ArmArch: "(arrRef + elemOffset) + index" // both ADD are BYREF here
+//
+// It'll allow us to hoist/CSE the invariant part which is "arrRef + elemOffset" on ArmArch.
+// We don't really need it on XArch because of more powerful addressing modes such as [base + index*scale + offset]
```

Comment:
I would like to understand how this impacts `ParseArrayAddress`: are there any diffs in what it returned before and what it returns now?

Reply:
ParseArrayAddressWork seems to be handling well this pattern and recognizes indices just fine for my samples, not sure about other methods in the diffs - there is too many of them